e2e/tests/admin_api/get_users_by_standard_attribute_test.go

```diff
@@ -0,0 +1,290 @@
+		t.Fatalf(err.Error())
+	}
+
+	resp, err := cmd.Client.GraphQLAPI(nil, nil, cmd.AppID, e2eclient.GraphQLAPIRequest{
+		Query: `
+			query {
+				getUsersByStandardAttribute(attributeName:"preferred_username", attributeValue: "e2e_admin_api_get_users_username") {
+						id
+						createdAt
+						standardAttributes
+						lastLoginAt
+				}
+			}
+
+		`,
+	})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if resp.Errors != nil {
+		t.Fatalf("error: %v", resp.Errors)
+	}
+
+	var doc map[string]any
+	if err := json.Unmarshal(resp.Data, &doc); err != nil {
+		panic(err)
+	}
+
+	result := doc["getUsersByStandardAttribute"].([]interface{})
+
+	if len(result) == 0 {
+		t.Fatal("result should not be empty")
+	}
+}
+
+func TestGetUsersByStandardAttributeNotFound(t *testing.T) {
+	cmd, err := testrunner.NewEnd2EndCmd(testrunner.NewEnd2EndCmdOptions{
+		TestCase: &testrunner.TestCase{
+			Path: "admin_api/get_users_by_standard_attribute_test.go",
+		},
+		Test: t,
+	})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// Prepare testing user
+	if err := cmd.ExecuteSQLInsertUpdateFile("../../tests/admin_api/get_users_admin_api_user.sql"); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	resp, err := cmd.Client.GraphQLAPI(nil, nil, cmd.AppID, e2eclient.GraphQLAPIRequest{
+		Query: `
+			query {
+				getUsersByStandardAttribute(attributeName:"email", attributeValue: "e2e_admin_api_get_users_not_found@example.com") {
+						id
+						createdAt
+						standardAttributes
+						lastLoginAt
+				}
+			}
+
+		`,
+	})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if resp.Errors != nil {
+		t.Fatalf("error: %v", resp.Errors)
+	}
+
+	var doc map[string]any
+	if err := json.Unmarshal(resp.Data, &doc); err != nil {
+		panic(err)
+	}
+
+	result := doc["getUsersByStandardAttribute"].([]interface{})
+
+	if len(result) > 0 {
+		t.Fatal("result should be empty")
+	}
+}
+
+func TestGetUsersByStandardAttributeInvalidAttributeValue(t *testing.T) {
+	cmd, err := testrunner.NewEnd2EndCmd(testrunner.NewEnd2EndCmdOptions{
+		TestCase: &testrunner.TestCase{
+			Path: "admin_api/get_users_by_standard_attribute_test.go",
+		},
+		Test: t,
+	})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// Prepare testing user
+	if err := cmd.ExecuteSQLInsertUpdateFile("../../tests/admin_api/get_users_admin_api_user.sql"); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	resp, err := cmd.Client.GraphQLAPI(nil, nil, cmd.AppID, e2eclient.GraphQLAPIRequest{
+		Query: `
+			query {
+				getUsersByStandardAttribute(attributeName:"phone_number", attributeValue: "e2e_admin_api_get_users@example.com") {
+						id
+						createdAt
+						standardAttributes
+						lastLoginAt
+				}
+			}
+
+		`,
+	})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if resp.Errors == nil {
+		t.Fatalf("expect error, got nil")
+	}
+
+	for _, err := range resp.Errors {
+		if err.Extensions.Reason == "GetUsersInvalidArgument" {
+			return
+		}
+	}
+
+	t.Fatalf("expect error GetUsersInvalidArgument, got %v", resp.Errors)
```

Comment:
We should use ShouldEqualJSON to match the actual response.

Reply:
updated to use `So` to assert value